app/controllers/application_controller.rb

```diff
@@ -1,10 +1,160 @@
 require './config/environment'
+require 'sinatra/base'
+require 'rack-flash'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
-
+  enable :sessions
+  use Rack::Flash
   configure do
+    set :session_secret, "secret"
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if is_logged_in?
+      redirect to '/tweets'
+    end
+
+    erb :"/users/create_user"
+  end
+
+  post '/signup' do
+    params.each do |label, input|
+      if input.empty?
+        flash[:new_user_error] = "Please enter a value for #{label}"
+        redirect to '/signup'
+      end
+    end
+
+    user = User.create(:username => params["username"], :email => params["email"], :password => params["password"])
+    session[:user_id] = user.id
+
+    redirect to '/tweets'
+  end
+
+  get '/login' do
+    if is_logged_in?
+      redirect to '/tweets'
+    end
+
+    erb :"/users/login"
+  end
+
+  post '/login' do
+    user = User.find_by(:username => params["username"])
+
+    if user && user.authenticate(params[:password])
+      # binding.pry
+      session[:user_id] = user.id
+      redirect '/tweets'
+    else
+      # binding.pry
+      flash[:login_error] = "Incorrect login. Please try again."
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets' do
+    if !is_logged_in?
+      redirect to '/login'
+    end
+    @tweets = Tweet.all
+    @user = current_user
+    # binding.pry
+    erb :'tweets/tweets'
+  end
+
+  get '/tweets/new' do
+    if !is_logged_in?
+      redirect to '/login'
+    end
+    erb :"/tweets/create_tweet"
+  end
+
+  post '/tweets' do
+    user = current_user
+    if params["content"].empty?
+      flash[:empty_tweet] = "Please enter content for your tweet"
+      redirect to '/tweets/new'
+    end
+    tweet = Tweet.create(:content => params["content"], :user_id => user.id)
+
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.